pkgs/tools/filesystems/gitfs/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, python, buildPythonPackage, pythonPackages }:
+
+let inherit (stdenv.lib) getVersion versionAtLeast versionOlder; in
+
+assert versionAtLeast (getVersion pythonPackages.pygit2) "0.21.0";
```

Comment:
Why this kind of versioning? If you need pygit2 between 0.21 and 0.22, then pass an appropriate pygit2 derivation.